File: 2348-number-of-zero-filled-subarrays/2348. Number of Zero-Filled Subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        long long res = 0;
        int l = 0, r = 0;
        vector<int> zeros;
        while (r < nums.size()) {
            int count = 0;
            while (r < nums.size() && nums[r] == 0) {
                r++;
                count++;
                if (r < nums.size() && nums[r] != 0) {
                    r--;
                    break;
                }
            }
            r++;
            if (count != 0) zeros.push_back(count);
        }
        for (long long num: zeros) {
            res += num * (num+1) / 2;
        }

        return res;
    }
};

File: 2348-number-of-zero-filled-subarrays/2348. Number of Zero-Filled Subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        res = 0
        curr = 0
        for n in nums:
            if n == 0:
                curr += 1
            else:
                res += curr * (curr + 1) // 2
                curr = 0
        if curr:
            res += curr * (curr + 1) // 2
        return res